### PR TITLE
Start publishing the cursed token on GitHub Pages

### LINKED ISSUE #5: publish token to GH Pages?
This is very neat (an abomination but a neat one) , I'm thinking of using this in my own testing as well... The only tricky part seems to be finding a token: see https://github.com/sigstore/sigstore-conformance/blob/6c958ac7c74146cb694b9bdb3f8111c3e447d6cf/test/conftest.py#L75 for an example

What if the workflow published to GitHub Pages with a static filename?  That way the newest produced token could always be available at https://sigstore-conformance.github.io/extremely-dangerous-public-oidc-beacon/cursed-token.txt. Am I missing anything?

The downloader would still have to deal with the token being too old (unsure how at this hand waving stage) but all of this would go away:
* GitHub auth
* workflow run json download
* workflow run json parsing
* unzipping

### PR DESCRIPTION
**Pull request by jku**
Start publishing the token on GitHub Pages (https://sigstore-conformance.github.io/extremely-dangerous-public-oidc-beacon/oidc-token.txt):
* This makes it a lot easier to find, download and use the token
* The old artifact upload is preserved so current scripts (so sigstore-conformance 0.7, 0.8) should keep working

Fixes #5.

Details:
* All known sigstore-conformance users have updated to v0.8 or are using main: I believe this is safe to merge.
* The GitHub Pages source is set to "Actions" in GH settings so this should start working right away. I can't see the environment settings but they should be correct by default (there should be a "github-pages" env and main branch should be allowed to deploy).
* The legacy artifact upload that is currently preserved can be removed once sigstore-conformance uses the new token location and we've seen that it's reliable.
* At that point I think the workflows in this project can be simplified significantly: the reason for the workflow dispatch dance is that published artifacts are not made available until the workflow finishes -- that limitation likely does not apply to Pages publishing (although it remains to be seen if GitHub is ok with publishing to Pages multiple times from the same workflow)

### CODE REVIEW

**Comment by jku** [comment]
> This is a draft until sigstore/sigstore-conformance#102 is merged

I supposed we should wait until there's a sigstore-conformance release and until the known users have upgraded as well.

**Comment by jku** [comment]
note to self: other conformance users have upgraded (or are using main branch), sigstore-js and sigstore-java are still on 0.6 at the time of this comment

**Comment by jku** [comment]
[This bit in sigstore-conformance](https://github.com/sigstore/sigstore-conformance/blob/6c958ac7c74146cb694b9bdb3f8111c3e447d6cf/test/conftest.py#L30C1-L30C36) I'm not sure about:

> _OIDC_BEACON_WORKFLOW_ID = 55399612

Does the workflow id stay the same? I expect that it does but I can't be sure.

**Comment by woodruffw** [comment]
> Does the workflow id stay the same? I expect that it does but I can't be sure.

I _think_ it does, but we can confirm/update with <https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#get-a-workflow> if it breaks after merging 🙂

**Review by woodruffw — APPROVED** [review]
LGTM! I'd like @tetsuo-cpp or @tnytown to also take a brief look at this if either have time 🙂

**Comment by woodruffw** [comment]
> In my experience, GitHub Pages sometimes takes a bit to propagate changes due to caching -- is this a concern w.r.t. the tokens, given their relatively short validity period?

Caching on the HTTP header side, or caching on the deployment side? I wouldn't be surprised if GHP serves HTTP caching headers, but we can always ignore those 🙂

**Comment by woodruffw** [comment]
Gotcha. I _think_ we can go ahead and deploy this as-is; if we run into consistent issues, then we'll look into another publication mechanism or location.

**Comment by woodruffw** [comment]
Appears to be working: https://sigstore-conformance.github.io/extremely-dangerous-public-oidc-beacon/oidc-token.txt

**Pull request merged**
